library/src/main/java/com/novoda/downloadmanager/NotificationCreator.java

```diff
@@ -47,8 +35,7 @@ public Notification getNotification() {
         };
     }
 
-    @RequiresApi(Build.VERSION_CODES.O)
-    NotificationChannel createNotificationChannel() {
-        return new NotificationChannel(channelId, userFacingChannelDescription, importance);
+    String channelId() {
+        return channelId;
```

Comment:
I'm missing something: where is this used now?
Can we not just use the value set to the channelProvider (thus requiring the client to specify it only once?)

Reply:
That would mean that I would need an instance of the provider here. Let me see what I can do for that.